---
layout: default
title: Home
nav_order: 1
description: "Just the Docs is a responsive Jekyll theme with built-in search that is easily customizable and hosted on GitHub Pages."
permalink: /
---

# Focus on writing good documentation
{: .fs-9 }

Just the Docs gives your documentation a jumpstart with a responsive Jekyll theme that is easily customizable and hosted on GitHub Pages.
{: .fs-6 .fw-300 }

[Get started now](#getting-started){: .btn .btn-primary .fs-5 .mb-4 .mb-md-0 .mr-2 } [View it on GitHub](https://github.com/pmarsceill/just-the-docs){: .btn .fs-5 .mb-4 .mb-md-0 }

---

## Getting started

### Shotgun Metagenomic Tools
#### Quality control of raw data
#### Remove human sequences
####  Quality control of trimmed data
####  De novo assembly
####  bining
####  Quality control of bining
####  Phylogenetic and taxonomic analyses
####  Functional characterization
####   Species prevalence and abundance
####   Comparative genomics
####   Multiomics assays
####  Correlation analysis
####  Referenses
1. Duvallet, C., Gibbons, S. M., Gurry, T., Irizarry, R. A. & Alm, E. J. Meta-analysis of gut microbiome studies identifies disease-specific and shared responses. Nat. Commun. 8, 1784 (2017).
2. Quince, C., Walker, A. W., Simpson, J. T., Loman, N. J. & Segata, N. Shotgun metagenomics, from sampling to analysis. Nat. Biotechnol. 35, 833–844 (2017).
3. Browne, H. P. et al. Culturing of ‘unculturable’ human microbiota reveals novel taxa and extensive sporulation. Nature 533, 543–546 (2016).
4. Forster, S. C. et al. A human gut bacterial genome and culture collection for precise and efficient metagenomic analysis. Nat. Biotechnol. 37, 186–192 (2019)
5. Kang, D. D., Froula, J., Egan, R. & Wang, Z. MetaBAT, an efficient tool for accurately reconstructing single genomes from complex microbial communities. PeerJ 3, e1165 (2015).
6. Parks, D. H., Imelfort, M., Skennerton, C. T., Hugenholtz, P. & Tyson, G. W. CheckM: assessing the quality of microbial genomes recovered from isolates,single cells, and metagenomes. Genome Res. 25, 1043–1055 (2015).
7. Li,D.,Liu,C.-M.,Luo,R.,Sadakane,K.&Lam,T.-W.MEGAHIT:anultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph. Bioinformatics 31, 1674–1676 (2015).
8. Uritskiy,G.V.,DiRuggiero,J.&Taylor,J.MetaWRAP-aflexiblepipelinefor genome-resolved metagenomic data analysis. Microbiome 6, 158 (2018).
9. Varghese,N.J.etal.Microbialspeciesdelineationusingwholegenome sequences. Nucleic Acids Res. 43, 6761–6771 (2015).
10. Segata, N., Börnigen, D., Morgan, X. C. & Huttenhower, C. PhyloPhlAn is a new method for improved phylogenetic and taxonomic placement of microbes. Nat. Commun. 4, 2304 (2013).
11. Blin,K.etal.antiSMASH4.0—improvementsinchemistryprediction and gene cluster boundary identification. Nucleic Acids Res. 45, W36–W41 (2017).
12. Mitchell,A.L.etal.EBIMetagenomicsin2017:enrichingtheanalysisof microbial communities, from sequence reads to assemblies. Nucleic Acids Res. 46, D726–D735 (2018).
13. Kanehisa,M.,Sato,Y.&Morishima,K.BlastKOALAandGhostKOALA:KEGG tools for functional characterization of genome and metagenome sequences. J. Mol. Biol. 428, 726–731 (2016).
14. Crichton,R.R.IronMetabolism:FromMolecularMechanismstoClinical Consequences. (John Wiley, Hoboken, NJ, 2016).
15. Pasolli,E.etal.Extensiveunexploredhumanmicrobiomediversityrevealedby over 150,000 genomes from metagenomes spanning age, geography, and lifestyle. Cell 176, 649–662 (2019).
16. Ren, J., Ahlgren, N. A., Lu, Y. Y., Fuhrman, J. A. & Sun, F. VirFinder: a novel k-mer based tool for identifying viral sequences from assembled metagenomic data. Microbiome 5, 69 (2017).
17. Markowitz, V. M. et al. IMG: the Integrated Microbial Genomes database and comparative analysis system. Nucleic Acids Res. 40, D115–D122 (2012).
18. Olm,M.R.,Brown,C.T.,Brooks,B.&Banfield,J.F.dRep:atoolforfastand accurate genomic comparisons that enables improved genome recovery from metagenomes through de-replication. ISME J. 11, 2864–2868 (2017).
19. Stamatakis,A.RAxMLversion8:atoolforphylogeneticanalysisand post-analysis of large phylogenies. Bioinformatics 30, 1312–1313 (2014).
20. Letunic,I.&Bork,P.Interactivetreeoflife(iTOL)v3:anonlinetoolforthedisplay and annotation of phylogenetic and other trees. Nucleic Acids Res. 44, W242–W245 (2016).
21. Revell, L. J. phytools: an R package for phylogenetic comparative biology (and other things). Methods Ecol. Evol. 3, 217–223 (2012).
22. Zmora, N., Zilberman-Schapira, G., Suez, J., Mor, U., Dori-Bachash, M., Bashiardes, S., et al. (2018). Personalized gut mucosal colonization resistance to empiric probiotics is associated with unique host and microbiome features. Cell 174:1388-1405.e21. doi: 10.1016/j.cell.2018.08.041
23. Liu, B., and Pop, M. (2009). ARDB–antibiotic resistance genes database. Nucleic Acids Res. 37, D443–D447. doi: 10.1093/nar/gkn656
24. Chen, L., Zheng, D., Liu, B., Yang, J., and Jin, Q. (2016). VFDB 2016: hierarchical and refined dataset for big data analysis–10 years on. Nucleic Acids Res. 44, D694–D697. doi: 10.1093/nar/gkv1239
25. 
####  Github
1. [A new genomic blueprint of the human gut microbiota] (https://github.com/Finn-Lab/MGS-gut)
2.  
####  Online tools
1. [Predictive metabolomic profiling of microbial communities using amplicon or metagenomic sequences]
 (http:// huttenhower.sph.harvard.edu/melonnpan/]




Just the Docs is built for [Jekyll](https://jekyllrb.com), a static site generator. View the [quick start guide](https://jekyllrb.com/docs/) for more information. Just the Docs requires no special plugins and can run on GitHub Pages' standard Jekyll compiler. The [Jekyll SEO Tag plugin](https://github.com/jekyll/jekyll-seo-tag) is included by default (no need to run any special installation) to inject SEO and open graph metadata on docs pages. For information on how to configure SEO and open graph metadata visit the [Jekyll SEO Tag usage guide](https://jekyll.github.io/jekyll-seo-tag/usage/).

### Quick start: Use as a GitHub Pages remote theme

1. Add Just the Docs to your Jekyll site's `_config.yml` as a [remote theme](https://blog.github.com/2017-11-29-use-any-theme-with-github-pages/)
```zsh
parallel -j 35 --xapply \
$'kneaddata -i {1} -i {2} \
-o kneaddata_res/{3} -v -t 20 --fastqc FastQC --remove-intermediate-output \
--bowtie2-options "--very-sensitive --dovetail" \
--reference-db /data3/zhaoxia/metagenomeTools/kneaddata_db/bowtie2_contam_hg37_db' \
::: /data2/ZhouZhiyuan/RJNA678737/metagenome/*_1.fastq.gz ::: /data2/ZhouZhiyuan/RJNA678737/metagenome/*_2.fastq.gz :::: samplelist
```
<small>You must have GitHub Pages enabled on your repo, one or more Markdown files, and a `_config.yml` file. [See an example repository](https://github.com/pmarsceill/jtd-remote)</small>

### Local installation: Use the gem-based theme

1. Install the Ruby Gem
```bash
$ for i in $(cat samplelist )
do
kneaddata -i /data2/ZhouZhiyuan/RJNA678737/metagenome/"$i"_1.fastq.gz -i /data2/ZhouZhiyuan/RJNA678737/metagenome/"$i"_2.fastq.gz --reference-db /data3/zhaoxia/metagenomeTools/kneaddata_db/bowtie2_contam_hg37_db -v -t 20 --fastqc FastQC --remove-intermediate-output --bowtie2-options "--very-sensitive --dovetail" -o ./kneaddata_res/"$i"
done
```

```bash
$ mkdir kraken_braken_merge_tables
python2 /usr/local/bin/combine_bracken_outputs.py --files ./bracken_res/*P.braken  -o ./kraken_braken_merge_tables/combine_phylum.txt
python2 /usr/local/bin/combine_bracken_outputs.py --files ./bracken_res/*C.braken  -o ./kraken_braken_merge_tables/combine_Class.txt
python2 /usr/local/bin/combine_bracken_outputs.py --files ./bracken_res/*O.braken  -o ./kraken_braken_merge_tables/combine_Order.txt
python2 /usr/local/bin/combine_bracken_outputs.py --files ./bracken_res/*G.braken  -o ./kraken_braken_merge_tables/combine_Genus.txt
python2 /usr/local/bin/combine_bracken_outputs.py --files ./bracken_res/*F.braken  -o ./kraken_braken_merge_tables/combine_Family.txt
python2 /usr/local/bin/combine_bracken_outputs.py --files ./bracken_res/*S.braken  -o ./kraken_braken_merge_tables/combine_Spices.txt
```

```zsh
# _Optional:
for i in $(cat samplelist)
 do
kraken2 --db /data3/zhaoxia/metagenomeTools/Kraken_database/krakenV1_db/ --threads 56  --report ./kraken_res/"$i".report --output ./kraken_res/"$i".output  ./clean_data/"$i"_kneaddata.fastq
bracken -d /data3/zhaoxia/metagenomeTools/Kraken_database/krakenV1_db/ -i ./kraken_res/"$i".report -o ./bracken_res/"$i".P.braken -w ./bracken_res/"$i".P.braken.report -r 150 -l P
bracken -d /data3/zhaoxia/metagenomeTools/Kraken_database/krakenV1_db/ -i ./kraken_res/"$i".report -o ./bracken_res/"$i".S.braken -w ./bracken_res/"$i".S.braken.report -r 150 -l S 
bracken -d /data3/zhaoxia/metagenomeTools/Kraken_database/krakenV1_db/ -i ./kraken_res/"$i".report -o ./bracken_res/"$i".F.braken -w ./bracken_res/"$i".F.braken.report -r 150 -l F
bracken -d /data3/zhaoxia/metagenomeTools/Kraken_database/krakenV1_db/ -i ./kraken_res/"$i".report -o ./bracken_res/"$i".G.braken -w ./bracken_res/"$i".G.braken.report -r 150 -l G
bracken -d /data3/zhaoxia/metagenomeTools/Kraken_database/krakenV1_db/ -i ./kraken_res/"$i".report -o ./bracken_res/"$i".C.braken -w ./bracken_res/"$i".C.braken.report -r 150 -l C
bracken -d /data3/zhaoxia/metagenomeTools/Kraken_database/krakenV1_db/ -i ./kraken_res/"$i".report -o ./bracken_res/"$i".O.braken -w ./bracken_res/"$i".O.braken.report -r 150 -l O

done
```
2. Add Just the Docs to your Jekyll site’s `_config.yml`
```zsh
$ parallel -j 42 --xapply \
'megahit -1 {1} -2 {2} \
--out-prefix {3} \
-o Assembly_megahit_res/{3}' \
::: kneaddata_clean/*_1.fastq ::: kneaddata_clean/*_2.fastq :::: samplelist
```
3. _Optional:_ Initialize search data (creates `search-data.json`)
```zsh
$ cat kneaddata_clean/*_1.fastq > merge_assembly_reads/ALL_READS_1.fastq
  cat kneaddata_clean/*_2.fastq > merge_assembly_reads/ALL_READS_2.fastq
  metawrap assembly -1 merge_assembly_reads/ALL_READS_1.fastq -2 merge_assembly_reads/ALL_READS_2.fastq -m 24 -t 8 \
  --metaspades -o merged_megahit_assembly
```

```zsh
$ conda activate py3
/data3/zhaoxia/metagenomeTools/quast/quast.py ./merged_megahit_assembly/assembly.contigs.fa -o Assembly_quast_evaluation/megahit-report
conda deactivate
```

```zsh
$ mkdir metaphlan_microAbundance
metaphlan2.py merged_megahit_assembly/assembly.contigs.fa --input_type fasta > ./metaphlan_microAbundance/merged_contigs_profiled_metagenome.txt
```

3. 解决metawrap binning 时 Cannot repair BWA paired reads that have different name
```zsh
$ for i in $(cat samplelist)
do
bbrename.sh in=kneaddata_clean/"$i"_1_kneaddata_paired_1.fastq \
in2=kneaddata_clean/"$i"_1_kneaddata_paired_2.fastq \
out=bbrename/"$i"_rename_1.fastq out2=bbrename/"$i"_rename_2.fastq 
done
```
4. Bin the co-assembly with three different algorithms with the Binning module
运行metawrap binning前出现以下错误
samtools: error while loading shared libraries: libncurses.so.6，所以用conda py3的环境

Following assembly with 

```bash
# .. or if you're using a Gemfile (bundler) 
$ conda activate py3
$ metawrap binning -o metawrap_initial_bining -t 96 \
-a merged_megahit_assembly/assembly.contigs.fa --metabat2 --maxbin2 --concoct bbrename/SRR*fastq
$ conda deactivate
```
The quality of each bin was evaluated with CheckM according to the level of genome completeness and contamination.
CheckM is unable to evaluate non-prokaryotic genomes


4. Metagenomic strain-level population genomics
   StrainPhlAn is a computational tool for tracking individual strains across large set of samples. The input of StrainPhlAn is a set of metagenomic samples and for each species, the output is a multiple sequence alignment (MSA) file of all species strains reconstructed directly from the samples. From this MSA, StrainPhlAn calls RAxML (or other phylogenetic tree builders) to build the phylogenetic tree showing the strain evolution of the sample strains. For each sample, StrainPhlAn extracts the strain of a specific species by merging and concatenating all reads mapped against that species markers in the MetaPhlAn2 database.
```bash
$ strainphlan.py -h  
$ find /data3/zhaoxia/ -name "mpa_v*"
```
4. Pathogen screening
Screening for presence of genes in short read samples
full genome shotgun sequencing

→ MetaMLST

      screen for presence of MLST genes or self-defined reference gene sequences

→ SRST2

      screen for presence of MLST genes or self-defined reference gene sequences

→ ARIBA

      screen for antibiotic resistance genes or MLST genes (local assembly based approach)

Screening for presence of genes in assembled contigs
→ ABRicate

      screening of contigs for antimicrobial resistance or virulence genes

Screening for bacterial strains in short read samples
full genome shotgun sequencing

→ PanPhlAn

      screen for new strains of a species (E. coli) and identify the strain specific gene set

1. Download the reference genomes
```bash
$ mkdir Alistipes
$ cd AListipes
$ panphlan_download_pangenome.py -i Alistipes_finegoldii -o ./
# or
$ wget -c https://www.dropbox.com/s/sgpnfk5ahlpo7wq/Alistipes_finegoldii.tar.bz2\?dl\=0
$ ls  -l Alistipes_finegoldii/
```
This will download and extract a folder containing :

Alistipes_finegoldii_pangenome_contigs.fna : all contigs from the reference genomes
Alistipes_finegoldii.X.bt2 : the bowtie2 indexes (6 files)
panphlan_Alistipes_finegoldii_annot.tsv : an annotation mapping tsv file, can be useful for further analysis of gene families of interest
Alistipes_finegoldii_pangenome.tsv : a pangenome tsv file containing information about gene families UniRef90 ID, gene family name and position (genome, contig, start and stop) like the following example.


5. Point your web browser to [http://localhost:4000](http://localhost:4000)

If you're hosting your site on GitHub Pages, [set up GitHub Pages and Jekyll locally](https://help.github.com/en/articles/setting-up-your-github-pages-site-locally-with-jekyll) so that you can more easily work in your development environment.

### Configure Just the Docs

- [See configuration options]({{ site.baseurl }}{% link docs/configuration.md %})

---

## About the project

Just the Docs is &copy; 2017-{{ "now" | date: "%Y" }} by [Patrick Marsceill](http://patrickmarsceill.com).

### License

Just the Docs is distributed by an [MIT license](https://github.com/pmarsceill/just-the-docs/tree/master/LICENSE.txt).

### Contributing

When contributing to this repository, please first discuss the change you wish to make via issue,
email, or any other method with the owners of this repository before making a change. Read more about becoming a contributor in [our GitHub repo](https://github.com/pmarsceill/just-the-docs#contributing).

#### Thank you to the contributors of Just the Docs!

<ul class="list-style-none">
{% for contributor in site.github.contributors %}
  <li class="d-inline-block mr-1">
     <a href="{{ contributor.html_url }}"><img src="{{ contributor.avatar_url }}" width="32" height="32" alt="{{ contributor.login }}"/></a>
  </li>
{% endfor %}
</ul>

### Code of Conduct

Just the Docs is committed to fostering a welcoming community.

[View our Code of Conduct](https://github.com/pmarsceill/just-the-docs/tree/master/CODE_OF_CONDUCT.md) on our GitHub repository.
