---
layout: default
title: Microbiome in ICIs
nav_order: 1
description: "Decoding gut micobiome for clinical success"
permalink: /
---

# Microbiome analysis
{: .fs-9 }

Decoding gut micobiome for clinical success.
{: .fs-6 .fw-300 }

[Get started now](#getting-started){: .btn .btn-primary .fs-5 .mb-4 .mb-md-0 .mr-2 } [View it on GitHub](https://github.com/zhaoxia413/Metagenome_ICI){: .btn .fs-5 .mb-4 .mb-md-0 }

---

## Getting started
### Microbiomic database
1. [RDP](http://rdp.cme.msu.edu/index.jsp) RibosomalDatabase Project
2. [SILVA](https://www.arb-silva.de/)
3. [Greengenes](http://greengenes.lbl.gov/)
4. [EzBioCloud](http://www.ezbiocloud.net/dashboard)
5. [PR2](https://figshare.com/articles/PR2_rRNA_gene_database/3803709) ProtistRibosomal Reference database
6. [FunGene](http://fungene.cme.msu.edu/) Functional Gene
### Shotgun Metagenomic
Metagenomics is the study of microbial communities in their original living places. Metagenome sequencing refers to sequencing the entire genomes of all microbes present in a sample in order to explore taxonomic, functional, and evolutionary aspects.

---

### Metagenomics sampling projects
1. Human Microbiome Project [HMP]()
2. Human Gastrointestinal Bacteria Culture Collection [HBC]()
   A comprehensive set of 737 whole-genome-sequenced bacterial isolates, representing 273 species (105 novel species) from 31 families found in the human gastrointestinal microbiota.
   [Forster, S.C., Kumar, N., Anonye, B.O. et al. A human gut bacterial genome and culture collection for improved metagenomic analyses. Nat Biotechnol 37, 186–192 (2019). https://doi.org/10.1038/s41587-018-0009-7](https://doi.org/10.1038/s41587-018-0009-7)
3. [Tara Oceans Expedition](https://doi.pangaea.de/10.1594/PANGAEA.840721)
   The present data set is a registry of samples from the Tara Oceans Expedition (2009-2013) that were selected for publication in a special issue of the SCIENCE journal (see related references below). The registry provides details about the sampling location and methodology of each sample. Uniform resource locators (URLs) offer direct links to additional contextual environmental data and to the corresponding sequence runs used for analysis in the related literature publications in the SCIENCE journal.
4. [American-Gut](https://github.com/biocore/American-Gut) 
   
   American Gut sequences and metadata are deposited in The European Bioinformatics Institute under the accession ERP012803.
   
   Bloom sequences found in the data repository are correct and up to date.
   
   OTU tables and mapping files hosted in this repository reflects the state of the project in May 2015 and before. This includes an earlier version of the American Gut survey and dietary questionnaire. Data in GitHub has been scrubbed for PHI. A listing of processed data with the new survey can be found at ftp://ftp.microbio.me/AmericanGut.
   
   The latest OTU tables and precalculated diversity comparisons generated by the primary processing notebook set can be found at ftp://ftp.microbio.me/AmericanGut/latest.

---

##  Shotgun Metagenomic Analysis

---
###  Data download
1. [Metagenomics toolkit](https://github.com/EBI-Metagenomics/emg-toolkit) enables scientists to download all of the sample metadata for a given study or sequence to a single csv file.
2. (ffq)(https://github.com/pachterlab/ffq).Fetch run information from the European Nucleotide Archive (ENA).
3. [NCBI Genome Downloading](https://github.com/kblin/ncbi-genome-download)
   ```zsh
   $ ncbi-genome-download --genera "Streptomyces coelicolor" bacteria
   ```
###  Quality control of raw data
###  Remove human sequences
1. [kneaddata](https://github.com/biobakery/kneaddata)
###  Quality control of trimmed data
###  De novo assembly
1. short reads assembler[SPAdes](http://bioinf.spbau.ru/spades)
2. for assembling large and complex metagenomics data[MEGAHIT](https://github.com/voutcn/megahit)
   MEGAHIT is an ultra-fast and memory-efficient NGS assembler. It is optimized for metagenomes, but also works well on generic single genome assembly (small or mammalian size) and single-cell assembly.
###  Alignment
1. [bowtie2](https://github.com/BenLangmead/bowtie2)
2. [BWA](https://github.com/lh3/bwa)
#### reads binning: from clean reads
总体来说应用最广泛的就是基于genes binning 和 contig binning
### Genes binning: from gene
Genes binning方法一般是在宏基因组做完组装和基因预测之后，把所有样品中预测到的基因混合在一起，去冗余得到unique genes集合，对这个unique genes集合进行binning，主要是根据gene在各个样品中的丰度变化模式，计算gene之间的相关性，利用这种相关性进行聚类。
1. metagenomic linkage groups ( MLG )
2. metagenomic clusters ( MGC )
3. metagenomic species ( MGS )
4. metagenomic operational taxonomic units ( MetaOTUs )

###  Contig bining: from contig
1. [MetaBin](http://metabin.riken.jp/)
2. [metaBAT](https://bitbucket.org/berkeleylab/metabat)
3. [CONCOCT](https://github.com/BinPro/CONCOCT)
4. [MetaAnnotator/MetaCluster] is a novel software for binning and annotating short paired-end reads(https://i.cs.hku.hk/~alse/MetaCluster/)
###  Quality control of bining
1. [CheckM](http://ecogenomics.github.io/CheckM/)
###  Phylogenetic and taxonomic analyses
1. [Kaiju](https://github.com/zhaoxia413/kaiju) is a program for the taxonomic classification of high-throughput sequencing reads, e.g., Illumina or Roche/454, from whole-genome sequencing of metagenomic DNA. Reads are directly assigned to taxa using the NCBI taxonomy and a reference database of protein sequences from microbial and viral genomes.
2. The second version of the Kraken taxonomic sequence classification system [kraken2](https://github.com/DerrickWood/kraken2)
3. 
###  Functional characterization
1. [The Resistance Gene Identifier (RGI)](https://github.com/arpcard/rgi)
2. [eggNOG-mapper](http://eggnogdb.embl.de/#/app/emapper)
3. [Predictive metabolomic profiling of microbial communities using amplicon or metagenomic sequences](http:// huttenhower.sph.harvard.edu/melonnpan/]
###  Species prevalence and abundance
1. [PanPhlAn 3 - strain detection and characterization](https://github.com/SegataLab/panphlan)

### Strain-level
1. Strain-level Metagenomic Estimation of Growth rate [SMEG](https://github.com/ohlab/SMEG) measures growth rates of bacterial subspecies or strains from complex metagenomic samples. SMEG is capable of identifying novel or uncharacterized strains in a given sample prior to growth estimation.
[Emiola, A., Zhou, W., Oh, J. (2020) "Metagenomic growth rate inferences of strains in situ," Science Advances, 6(17), p.eaaz2299](https://advances.sciencemag.org/content/6/17/eaaz2299)

### Phylogenetic tree
1. [Pairdist](https://github.com/frederic-mahe/pairdist)：用于建NJ树
2. [TreeBest]（https://github.com/lh3/treebest 或者 http://treesoft.sourceforge.net/)
   [TreeBest的使用](http://blog.sina.com.cn/s/blog_620b35790100mcp6.html)
3. [Fasttree](http://www.microbesonline.org/fasttree/)
4. [RAxML](https://sco.h-its.org/exelixis/web/software/raxml/index.html)：ML树工具
5. [PhyML](http://www.atgc-montpellier.fr/phyml/）：在线构建ML树的工具，也可以本地执行
6. [profileNJ](https://github.com/maclandrol/profileNJ)：使用物种数和NJ树校正Gene tree
7. [Figtree](http://tree.bio.ed.ac.uk/software/figtree/)：a graphical viewer of phylogenetic trees and as a program for producing publication-ready figures.
8. [Dendroscope](http://dendroscope.org/)：Software for visualizing phylogenetic trees and rooted networks.
9. [PATRIC](https://www.patricbrc.org/)：Phylogenetic Tree Builder
10. [TempEst](http://tree.bio.ed.ac.uk/software/tempest/) TempEst is a tool for investigating the temporal signal and 'clocklikeness' of molecular phylogenies.

###  Comparative genomics
1. [PGCGAP - the Prokaryotic Genomics and Comparative Genomics Analysis Pipeline](https://github.com/liaochenlanruo/pgcgap)
2. [PanViz](https://github.com/thomasp85/PanViz) is a comprehensive interactive data visualization tool for investigating and understanding comparative microbial genomics data (often termed pangenomes). It is build around the idea of a dependency-free html file that makes it as easy to share with co-workers without installation of dedicated programs etc.
   [Usage](https://www.bioconductor.org/packages/release/bioc/vignettes/PanVizGenerator/inst/doc/panviz_howto.html)
3. [PanVizGenerator](https://github.com/thomasp85/PanVizGenerator) This R package is a companion to the PanViz javascript visualization for functionally annotated pangenomes. While PanViz is fully self-contained once it is created, this package takes care of converting your pangenome data into a PanViz file.
4. [FindMyFriends](http://bioconductor.org/packages/release/bioc/html/FindMyFriends.html) Microbial Comparative Genomics in R.
5. [Sibelia](http://bioinf.spbau.ru/en/sibelia): A comparative genomics tool
###  Multiomics assays
###  Correlation analysis
###  Piplines
1. [metaWrap](https://github.com/bxlab/metaWRAP)
2. [gatk4-pathseq](https://github.com/gatk-workflows/gatk4-pathseq) This repo contains workflows for computational pathogen discovery using PathSeq, a pipeline in the Genome Analysis Toolkit (GATK) for detecting microbial organisms in short-read deep sequencing samples taken from a host organism.

### pan-genomes analysis
1. [Roary](http://sanger-pathogens.github.io/Roary/) Large-scale prokaryote pan genome analysis Roary
2. [BPGA](https://iicb.res.in/bpga/index.html) is an ultra-fast software package that provides comprehensive pan genome analysis of microorganisms.PGA is an ultra-fast software package that provides comprehensive pan genome analysis of microorganisms. In addition to all types of routine pan genomic analyses (Pan genome Profiles, Pan/Core Phylogeny etc.), BPGA includes a number of novel downstream analysis features like Exclusive Gene Family Analysis, Atypical GC Content Analysis, Subset Analysis, MLST based on housekeeping genes and KEGG Distribution etc.
3. [PanGP](https://pangp.zhaopage.com/) is a tool for quickly analyzing bacterial pan-genome profile
4. [panOCT](https://sourceforge.net/projects/panoct/?source=directory) Pan-genome Ortholog Clustering Tool, is a program written in PERL for pan-genomic analysis of closely related prokaryotic species or strains. Unlike traditional graph-based ortholog detection programs, it uses micro synteny or conserved gene neighborhood (CGN) in addition to homology to accurately place proteins into orthologous clusters.
5. [PGAP](https://github.com/kastman/pgap-docker https://sourceforge.net/projects/pgap/) pan-genomes analysis pipeline. 太耗时了！！慎用！
6. [metaPGAP](https://github.com/mitul-patel/metaPGAP) metagenomic Pan Genome Analysis Pipeline
7. [PGAP-X](https://pgapx.ybzhao.com/) is a microbial comparative genomic analysis platform with graphic interface

### Antimicrobial and virulence genes 
1. [ABRicate](https://github.com/tseemann/abricate) Mass screening of contigs for antimicrobial resistance or virulence genes. It comes bundled with multiple databases: NCBI, CARD, ARG-ANNOT, Resfinder, MEGARES, EcOH, PlasmidFinder, Ecoli_VF and VFDB.
2. [ARIBA](https://github.com/sanger-pathogens/ariba) Antimicrobial Resistance Identification By Assembly
3. [Meta-MARC](https://github.com/lakinsm/meta-marc) Metagenomic Markov models for Antimicrobial Resistance Characterization
4. [DeepARG](https://bench.cs.vt.edu/deeparg) The deepARG models have been designed for computational analysis of next generation sequencing data such as Metagenomes. The main contribution of the deepARG models are their low false negative rate during predictions. Also, the gene-like sequences model is designed to find novel ARGs based on the sequence homology.
5. [BacMet](http://bacmet.biomedicine.gu.se/)is an easy-to-use bioinformatics resource of antibacterial biocide- and metal-resistance genes. BacMet consists of two databases:
A manually curated database of genes with experimentally confirmed resistance function
A database of predicted resistance genes based on sequence similarity to genes with experimentally confirmed function
6. [CARD](https://card.mcmaster.ca/) The Comprehensive Antibiotic Resistance Database
7. [SRST2](https://github.com/katholt/srst2) Short Read Sequence Typing for Bacterial Pathogens
This program is designed to take Illumina sequence data, a MLST database and/or a database of gene sequences (e.g. resistance genes, virulence genes, etc) and report the presence of STs and/or reference genes.
8. [Resfams](http://www.dantaslab.org/resfams)is a curated database of protein families and associated profile hidden Markov models (HMMs), confirmed for antibiotic resistance function and organized by ontology.
9. [NCBI Bacterial Antimicrobial Resistance Reference Gene Database](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA313047)
10. [ResFinder](https://cge.cbs.dtu.dk/services/ResFinder/) identifies acquired genes and/or finds chromosomal mutations mediating antimicrobial resistance in total or partial DNA sequence of bacteria.
11. [c-SSTAR](https://github.com/chrisgulvik/c-SSTAR) LI-Sequence Search Tool for Antimicrobial Resistance
12. [ARGs-OAP Galaxy](https://smile.hku.hk/SARGs) Online Analysis Pipeline for Anti-biotic Resistance Genes Detection from Meta-genomic Data Using an Integrated Structured ARG-database
    
    'Availible Tools':

    → SARGFAM: Sargfam database and profile Hidden Markov Models based alignment algorithm for ARGs-like sequences annotation were a function module in ARGs-OAP v2.0, supporting supplementary sequence retrieval strategy to current similarity search method (i.e. UBLAST and BLAST), especially applied on detecting remote homolog (novel genes).

    → ARGPORE: This tool can help users to identify ARG hits on nanopore reads and simultaneously find the carrier population of the identified ARGs by searching against phylogenetic marker genes on nanopore reads.

    → Microbial Source Tracking (MST): Microbial Source Tracking is a source-tracking platform using machine-learning classification with ARG abundance profiles. In other words, it precisely tracks antibiotic resistance gene pollution from different sources when your samples are related to human feces, animal feces, wastewater treatment plants, natural environments. This is a novel application of your high-throughput sequencing dataset. The developed source-tracking platform when coupling with proper experiment design and efficient metagenomic analysis tools will have significant implications for assessing AMR pollution. Following predicted source contribution status, risk ranking of different sources in ARG dissemination will be possible, thereby paving the way for establishing priority in mitigating ARG spread and designing effective control strategies.

    → ARGs-OSP: Antibiotic Resistant Genes Online Searching Platform. ARGs-OSP is online search platfrom that provides a global ARG profile covering the information of their phylogenetic and ecological distribution. Search and download functionality are designed for users to retrieve the occurrence of ARGs in different taxonomy and the abundance of ARGs in different habitats. Through data sharing, ARGs-OSP aims to motivate and facilitate future studies into mining new information and knowledge from the combined data, without making repeated efforts in dataset processing.

    → I-VIP: Integron Visualization and Identification Pipeline: 


####  Github
1. [A new genomic blueprint of the human gut microbiota](https://github.com/Finn-Lab/MGS-gut)

### Quick start: Use as a GitHub Pages remote theme

1. Add Just the Docs to your Jekyll site's `_config.yml` as a [remote theme](https://blog.github.com/2017-11-29-use-any-theme-with-github-pages/)
   
```zsh
parallel -j 35 --xapply \
$'kneaddata -i {1} -i {2} \
-o kneaddata_res/{3} -v -t 20 --fastqc FastQC --remove-intermediate-output \
--bowtie2-options "--very-sensitive --dovetail" \
--reference-db /data3/zhaoxia/metagenomeTools/kneaddata_db/bowtie2_contam_hg37_db' \
::: /data2/ZhouZhiyuan/RJNA678737/metagenome/*_1.fastq.gz ::: /data2/ZhouZhiyuan/RJNA678737/metagenome/*_2.fastq.gz :::: samplelist
```
<small>You must have GitHub Pages enabled on your repo, one or more Markdown files, and a `_config.yml` file. [See an example repository](https://github.com/pmarsceill/jtd-remote)</small>

### Local installation: Use the gem-based theme

1. Install the Ruby Gem
   
```bash
$ for i in $(cat samplelist )
do
kneaddata -i /data2/ZhouZhiyuan/RJNA678737/metagenome/"$i"_1.fastq.gz -i /data2/ZhouZhiyuan/RJNA678737/metagenome/"$i"_2.fastq.gz --reference-db /data3/zhaoxia/metagenomeTools/kneaddata_db/bowtie2_contam_hg37_db -v -t 20 --fastqc FastQC --remove-intermediate-output --bowtie2-options "--very-sensitive --dovetail" -o ./kneaddata_res/"$i"
done
```

```bash
$ mkdir kraken_braken_merge_tables
python2 /usr/local/bin/combine_bracken_outputs.py --files ./bracken_res/*P.braken  -o ./kraken_braken_merge_tables/combine_phylum.txt
python2 /usr/local/bin/combine_bracken_outputs.py --files ./bracken_res/*C.braken  -o ./kraken_braken_merge_tables/combine_Class.txt
python2 /usr/local/bin/combine_bracken_outputs.py --files ./bracken_res/*O.braken  -o ./kraken_braken_merge_tables/combine_Order.txt
python2 /usr/local/bin/combine_bracken_outputs.py --files ./bracken_res/*G.braken  -o ./kraken_braken_merge_tables/combine_Genus.txt
python2 /usr/local/bin/combine_bracken_outputs.py --files ./bracken_res/*F.braken  -o ./kraken_braken_merge_tables/combine_Family.txt
python2 /usr/local/bin/combine_bracken_outputs.py --files ./bracken_res/*S.braken  -o ./kraken_braken_merge_tables/combine_Spices.txt
```

```zsh
# _Optional:
for i in $(cat samplelist)
 do
kraken2 --db /data3/zhaoxia/metagenomeTools/Kraken_database/krakenV1_db/ --threads 56  --report ./kraken_res/"$i".report --output ./kraken_res/"$i".output  ./clean_data/"$i"_kneaddata.fastq
bracken -d /data3/zhaoxia/metagenomeTools/Kraken_database/krakenV1_db/ -i ./kraken_res/"$i".report -o ./bracken_res/"$i".P.braken -w ./bracken_res/"$i".P.braken.report -r 150 -l P
bracken -d /data3/zhaoxia/metagenomeTools/Kraken_database/krakenV1_db/ -i ./kraken_res/"$i".report -o ./bracken_res/"$i".S.braken -w ./bracken_res/"$i".S.braken.report -r 150 -l S 
bracken -d /data3/zhaoxia/metagenomeTools/Kraken_database/krakenV1_db/ -i ./kraken_res/"$i".report -o ./bracken_res/"$i".F.braken -w ./bracken_res/"$i".F.braken.report -r 150 -l F
bracken -d /data3/zhaoxia/metagenomeTools/Kraken_database/krakenV1_db/ -i ./kraken_res/"$i".report -o ./bracken_res/"$i".G.braken -w ./bracken_res/"$i".G.braken.report -r 150 -l G
bracken -d /data3/zhaoxia/metagenomeTools/Kraken_database/krakenV1_db/ -i ./kraken_res/"$i".report -o ./bracken_res/"$i".C.braken -w ./bracken_res/"$i".C.braken.report -r 150 -l C
bracken -d /data3/zhaoxia/metagenomeTools/Kraken_database/krakenV1_db/ -i ./kraken_res/"$i".report -o ./bracken_res/"$i".O.braken -w ./bracken_res/"$i".O.braken.report -r 150 -l O
done
```
2. Add Just the Docs to your Jekyll site’s `_config.yml`
```zsh
$ parallel -j 42 --xapply \
'megahit -1 {1} -2 {2} \
--out-prefix {3} \
-o Assembly_megahit_res/{3}' \
::: kneaddata_clean/*_1.fastq ::: kneaddata_clean/*_2.fastq :::: samplelist
```
3. _Optional:_ Initialize search data (creates `search-data.json`)
```zsh
$ cat kneaddata_clean/*_1.fastq > merge_assembly_reads/ALL_READS_1.fastq
  cat kneaddata_clean/*_2.fastq > merge_assembly_reads/ALL_READS_2.fastq
  metawrap assembly -1 merge_assembly_reads/ALL_READS_1.fastq -2 merge_assembly_reads/ALL_READS_2.fastq -m 24 -t 8 \
  --metaspades -o merged_megahit_assembly
```

```zsh
$ conda activate py3
/data3/zhaoxia/metagenomeTools/quast/quast.py ./merged_megahit_assembly/assembly.contigs.fa -o Assembly_quast_evaluation/megahit-report
conda deactivate
```

```zsh
$ mkdir metaphlan_microAbundance
metaphlan2.py merged_megahit_assembly/assembly.contigs.fa --input_type fasta > ./metaphlan_microAbundance/merged_contigs_profiled_metagenome.txt
```

3. 解决metawrap binning 时 Cannot repair BWA paired reads that have different name
```zsh
$ for i in $(cat samplelist)
do
bbrename.sh in=kneaddata_clean/"$i"_1_kneaddata_paired_1.fastq \
in2=kneaddata_clean/"$i"_1_kneaddata_paired_2.fastq \
out=bbrename/"$i"_rename_1.fastq out2=bbrename/"$i"_rename_2.fastq 
done
```
4. Bin the co-assembly with three different algorithms with the Binning module
运行metawrap binning前出现以下错误
samtools: error while loading shared libraries: libncurses.so.6，所以用conda py3的环境

Following assembly with 

```bash
# .. or if you're using a Gemfile (bundler) 
$ conda activate py3
$ metawrap binning -o metawrap_initial_bining -t 96 \
-a merged_megahit_assembly/assembly.contigs.fa --metabat2 --maxbin2 --concoct bbrename/SRR*fastq
$ conda deactivate
```
The quality of each bin was evaluated with CheckM according to the level of genome completeness and contamination.
CheckM is unable to evaluate non-prokaryotic genomes


4. Metagenomic strain-level population genomics
   StrainPhlAn is a computational tool for tracking individual strains across large set of samples. The input of StrainPhlAn is a set of metagenomic samples and for each species, the output is a multiple sequence alignment (MSA) file of all species strains reconstructed directly from the samples. From this MSA, StrainPhlAn calls RAxML (or other phylogenetic tree builders) to build the phylogenetic tree showing the strain evolution of the sample strains. For each sample, StrainPhlAn extracts the strain of a specific species by merging and concatenating all reads mapped against that species markers in the MetaPhlAn2 database.
```bash
$ strainphlan.py -h  
$ find /data3/zhaoxia/ -name "mpa_v*"
```
4. Pathogen screening
Screening for presence of genes in short read samples
full genome shotgun sequencing

→ MetaMLST

      screen for presence of MLST genes or self-defined reference gene sequences

→ SRST2

      screen for presence of MLST genes or self-defined reference gene sequences

→ ARIBA

      screen for antibiotic resistance genes or MLST genes (local assembly based approach)

Screening for presence of genes in assembled contigs
→ ABRicate

      screening of contigs for antimicrobial resistance or virulence genes

Screening for bacterial strains in short read samples
full genome shotgun sequencing

→ PanPhlAn

      screen for new strains of a species (E. coli) and identify the strain specific gene set

1. Download the reference genomes
```bash
$ mkdir Alistipes
$ cd AListipes
$ panphlan_download_pangenome.py -i Alistipes_finegoldii -o ./
# or
$ wget -c https://www.dropbox.com/s/sgpnfk5ahlpo7wq/Alistipes_finegoldii.tar.bz2\?dl\=0
$ ls  -l Alistipes_finegoldii/
```
This will download and extract a folder containing :

Alistipes_finegoldii_pangenome_contigs.fna : all contigs from the reference genomes
Alistipes_finegoldii.X.bt2 : the bowtie2 indexes (6 files)
panphlan_Alistipes_finegoldii_annot.tsv : an annotation mapping tsv file, can be useful for further analysis of gene families of interest
Alistipes_finegoldii_pangenome.tsv : a pangenome tsv file containing information about gene families UniRef90 ID, gene family name and position (genome, contig, start and stop) like the following example.


5. Point your web browser to [http://localhost:4000](http://localhost:4000)

If you're hosting your site on GitHub Pages, [set up GitHub Pages and Jekyll locally](https://help.github.com/en/articles/setting-up-your-github-pages-site-locally-with-jekyll) so that you can more easily work in your development environment.

### Configure Just the Docs

- [See configuration options]({{ site.baseurl }}{% link docs/configuration.md %})

---

## About the project

Just the Docs is &copy; 2017-{{ "now" | date: "%Y" }} by [Patrick Marsceill](http://patrickmarsceill.com).

### License

Just the Docs is distributed by an [MIT license](https://github.com/pmarsceill/just-the-docs/tree/master/LICENSE.txt).

### Contributing

When contributing to this repository, please first discuss the change you wish to make via issue,
email, or any other method with the owners of this repository before making a change. Read more about becoming a contributor in [our GitHub repo](https://github.com/pmarsceill/just-the-docs#contributing).

#### Thank you to the contributors of Just the Docs!

<ul class="list-style-none">
{% for contributor in site.github.contributors %}
  <li class="d-inline-block mr-1">
     <a href="{{ contributor.html_url }}"><img src="{{ contributor.avatar_url }}" width="32" height="32" alt="{{ contributor.login }}"/></a>
  </li>
{% endfor %}
</ul>

### Code of Conduct

Just the Docs is committed to fostering a welcoming community.

[View our Code of Conduct](https://github.com/pmarsceill/just-the-docs/tree/master/CODE_OF_CONDUCT.md) on our GitHub repository.

###  Referenses

<small>
   1. Duvallet, C., Gibbons, S. M., Gurry, T., Irizarry, R. A. & Alm, E. J. Meta-analysis of gut microbiome studies identifies disease-specific and shared responses. Nat. Commun. 8, 1784 (2017).
   2. Quince, C., Walker, A. W., Simpson, J. T., Loman, N. J. & Segata, N. Shotgun metagenomics, from sampling to analysis. Nat. Biotechnol. 35, 833–844 (2017).
   3. Browne, H. P. et al. Culturing of ‘unculturable’ human microbiota reveals novel taxa and extensive sporulation. Nature 533, 543–546 (2016).
   4. Forster, S. C. et al. A human gut bacterial genome and culture collection for precise and efficient metagenomic analysis. Nat. Biotechnol. 37, 186–192 (2019)
   5. Kang, D. D., Froula, J., Egan, R. & Wang, Z. MetaBAT, an efficient tool for accurately reconstructing single genomes from complex microbial communities. PeerJ 3, e1165 (2015).
   6. Parks, D. H., Imelfort, M., Skennerton, C. T., Hugenholtz, P. & Tyson, G. W. CheckM: assessing the quality of microbial genomes recovered from isolates,single cells, and metagenomes. Genome Res. 25, 1043–1055 (2015).
   7.  Li,D.,Liu,C.-M.,Luo,R.,Sadakane,K.&Lam,T.-W.MEGAHIT:anultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph. Bioinformatics 31, 1674–1676 (2015).
   8.  Uritskiy,G.V.,DiRuggiero,J.&Taylor,J.MetaWRAP-aflexiblepipelinefor genome-resolved metagenomic data analysis. Microbiome 6, 158 (2018).
   9.  Varghese,N.J.etal.Microbialspeciesdelineationusingwholegenome sequences. Nucleic Acids Res. 43, 6761–6771 (2015).
   10. Segata, N., Börnigen, D., Morgan, X. C. & Huttenhower, C. PhyloPhlAn is a new method for improved phylogenetic and taxonomic placement of microbes. Nat. Commun. 4, 2304 (2013).
   11. Blin,K.etal.antiSMASH4.0—improvementsinchemistryprediction and gene cluster boundary identification. Nucleic Acids Res. 45, W36–W41 (2017).
   12.  Mitchell,A.L.etal.EBIMetagenomicsin2017:enrichingtheanalysisof microbial communities, from sequence reads to assemblies. Nucleic Acids Res. 46, D726–D735 (2018).
   13.  Kanehisa,M.,Sato,Y.&Morishima,K.BlastKOALAandGhostKOALA:KEGG tools for functional characterization of genome and metagenome sequences. J. Mol. Biol. 428, 726–731 (2016).
   14.  Crichton,R.R.IronMetabolism:FromMolecularMechanismstoClinical Consequences. (John Wiley, Hoboken, NJ, 2016).
   15.  Pasolli,E.etal.Extensiveunexploredhumanmicrobiomediversityrevealedby over 150,000 genomes from metagenomes spanning age, geography, and lifestyle. Cell 176, 649–662 (2019).
   16.  Ren, J., Ahlgren, N. A., Lu, Y. Y., Fuhrman, J. A. & Sun, F. VirFinder: a novel k-mer based tool for identifying viral sequences from assembled metagenomic data. Microbiome 5, 69 (2017).
   17.  Markowitz, V. M. et al. IMG: the Integrated Microbial Genomes database and comparative analysis system. Nucleic Acids Res. 40, D115–D122 (2012).
   18.  Olm,M.R.,Brown,C.T.,Brooks,B.&Banfield,J.F.dRep:atoolforfastand accurate genomic comparisons that enables improved genome recovery from metagenomes through de-replication. ISME J. 11, 2864–2868 (2017).
   19.  Stamatakis,A.RAxMLversion8:atoolforphylogeneticanalysisand post-analysis of large phylogenies. Bioinformatics 30, 1312–1313 (2014).
   20.  Letunic,I.&Bork,P.Interactivetreeoflife(iTOL)v3:anonlinetoolforthedisplay and annotation of phylogenetic and other trees. Nucleic Acids Res. 44, W242–W245 (2016).
   21.  Revell, L. J. phytools: an R package for phylogenetic comparative biology (and other things). Methods Ecol. Evol. 3, 217–223 (2012).
   22.  Zmora, N., Zilberman-Schapira, G., Suez, J., Mor, U., Dori-Bachash, M., Bashiardes, S., et al. (2018). Personalized gut mucosal colonization resistance to empiric probiotics is associated with unique host and microbiome features. Cell 174:1388-1405.e21. doi: 10.1016/j.cell.2018.08.041.
   23.  Liu, B., and Pop, M. (2009). ARDB–antibiotic resistance genes database. Nucleic Acids Res. 37, D443–D447. doi: 10.1093/nar/gkn656.
   24.   Chen, L., Zheng, D., Liu, B., Yang, J., and Jin, Q. (2016). VFDB 2016: hierarchical and refined dataset for big data analysis–10 years on. Nucleic Acids Res. 44, D694–D697. doi: 10.1093/nar/gkv1239.
   </small>